cidc_api/services/files.py

```diff
@@ -1,15 +1,70 @@
 """Endpoints for downloadable file operations"""
-
+import json
 from multiprocessing.pool import ThreadPool
 
 from eve import Eve
+from flask import Request, _request_ctx_stack
+from werkzeug.exceptions import BadRequest
+from werkzeug.datastructures import ImmutableMultiDict
 
 import gcloud_client
+from models import Users, Permissions
 
 
 def register_files_hooks(app: Eve):
     #     app.on_fetched_resource_downloadable_files += insert_download_urls
     app.on_fetched_item_downloadable_files += insert_download_url
+    app.on_pre_GET_downloadable_files = update_file_filters
+
+
+def update_file_filters(request: Request, _):
+    """
+    Modify a downloadable_files lookup to filter out files 
+    a user doesn't have permission to see.
+    """
+    user = _request_ctx_stack.top.current_user
+
+    permissions = Permissions.find_for_user(user)
+
+    # User cannot access any trials, so make filter guaranteed-empty
+    if len(permissions) == 0:
+        guaranteed_empty = "(trial==a and trial==b)"
+        lookup = request.args.copy()
+        lookup["where"] = guaranteed_empty
+        request.args = ImmutableMultiDict(lookup)
+        return
+
+    # Build a where-query that looks up only the downloadable_files with
+    # trial ID and assay type that the current user is allowed to view.
+    # If the user has permission to view WES for Trial "1" and Olink for "2",
+    # this query will look like:
+    #   (trial==1 and assay_type==wes)or(trial==2 and assay_type==olink)
+    where_query = "or".join(
+        f'(trial=="{p.trial_id}" and assay_type=="{p.assay_type}")' for p in permissions
```

Comment:
There's a possibility for query injection here
like 
```
> print('THIS="{that}"'.format(that='some" or THIS="anything'))
THIS="some" or THIS="anything"
```
 
```suggestion
        f'(trial=={p.trial_id!r} and assay_type=={p.assay_type!r})' for p in permissions
```